### ISSUE TITLE
nspawn: --network-interface fails with "Network interface is not initialized yet." on network interfaces created with iproute2

### ISSUE THREAD

**Issue by daandemeyer** [systemd/systemd#14223]
**systemd version the issue has been seen with**
> systemd-v244-rc1

**Used distribution**
 > Ubuntu Focal

**Expected behaviour you didn't see**
 > systemd-nspawn starts the container and inherits the veth interface created with iproute2 

**Unexpected behaviour you saw**
 > systemd-nspawn fails with "Network interface is not initialized yet."

**Steps to reproduce the problem**

```
ip link add name vi1 type veth peer name vi2
systemd-nspawn -b -D <rootfs> --network-interface vi1
```

**Comment by yuwata** [comment]
Please run with `SYSTEMD_LOG_LEVEL=debug` and provide the debugging logs.

**Comment by DaanDeMeyer** [comment]
```
Setting RLIMIT_CPU to infinity.                                                                                                                                                                                   
Setting RLIMIT_FSIZE to infinity.                                                                                                                                                                                 
Setting RLIMIT_DATA to infinity.                                                                                                                                                                                  
Setting RLIMIT_STACK to 8388608:infinity.                                                                                                                                                                         
Setting RLIMIT_CORE to 0:infinity.                                                                                                                                                                                
Setting RLIMIT_RSS to infinity.                                                                                                                                                                                   
Setting RLIMIT_NPROC to 1048576.                                                                                                                                                                                  
Setting RLIMIT_NOFILE to 1024:4096.                                                                                                                                                                               
Setting RLIMIT_MEMLOCK to 65536.                                                                                                                                                                                  
Setting RLIMIT_AS to infinity.                                                                                                                                                                                    
Setting RLIMIT_LOCKS to infinity.                                                                                                                                                                                 
Setting RLIMIT_SIGPENDING to 127502.                                                                                                                                                                              
Setting RLIMIT_MSGQUEUE to 819200.                                                                                                                                                                                
Setting RLIMIT_NICE to 0.                                                                                                                                                                                         
Setting RLIMIT_RTPRIO to 0.                                                                                                                                                                                       
Setting RLIMIT_RTTIME to infinity.                                                                                                                                                                                
Found cgroup on /sys/fs/cgroup/systemd, legacy hierarchy                                                                                                                                                          
Spawning container rootfs on /root/build/host/rootfs.                                                                                                                                                   
Press ^] three times within 1s to kill container.                                                                                                                                                                 
Outer child is initializing.                                                                                                                                                                                      
Mounting / (MS_REC|MS_SLAVE "")...                                                                                                                                                                                
Bind-mounting /root/build/host/rootfs on /root/build/host/rootfs (MS_BIND|MS_REC "")...                                                                                                       
Using legacy hierarchy for container.                                                                                                                                                                             
Mounting /root/build/host/rootfs (MS_REC|MS_SHARED "")...
Mounting tmpfs on /root/build/host/rootfs/tmp (MS_NOSUID|MS_NODEV|MS_STRICTATIME "mode=1777,uid=0,gid=0")...
Mounting tmpfs on /root/build/host/rootfs/sys (MS_NOSUID|MS_NODEV|MS_NOEXEC "mode=555,uid=0,gid=0")...
Mounting tmpfs on /root/build/host/rootfs/dev (MS_NOSUID|MS_STRICTATIME "mode=755,uid=0,gid=0")...
Mounting tmpfs on /root/build/host/rootfs/dev/shm (MS_NOSUID|MS_NODEV|MS_STRICTATIME "mode=1777,uid=0,gid=0")...
Mounting tmpfs on /root/build/host/rootfs/run (MS_NOSUID|MS_NODEV|MS_STRICTATIME "mode=755,uid=0,gid=0")...
Bind-mounting /sys/fs/selinux on /root/build/host/rootfs/sys/fs/selinux (MS_BIND "")...
Failed to mount /sys/fs/selinux (type n/a) on /root/build/host/rootfs/sys/fs/selinux (MS_BIND ""): No such file or directory
Remounting /root/build/host/rootfs/sys/fs/selinux (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Failed to mount n/a (type n/a) on /root/build/host/rootfs/sys/fs/selinux (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND ""): Invalid argument
Mounting devpts on /root/build/host/rootfs/dev/pts (MS_NOSUID|MS_NOEXEC "newinstance,ptmxmode=0666,mode=620,gid=5")...
Bind-mounting /run/systemd/nspawn/propagate/rootfs on /root/build/host/rootfs/run/systemd/nspawn/incoming (MS_BIND "")...
Remounting /root/build/host/rootfs/run/systemd/nspawn/incoming (MS_RDONLY|MS_REMOUNT|MS_BIND "")...
Mounting /root/build/host/rootfs/run/systemd/nspawn/incoming (MS_SLAVE "")...
Kernel keyring access prohibited, ignoring.
Failed to remove '/root/build/host/rootfs/etc/localtime', ignoring: No such file or directory
Inner child is initializing.
Mounting proc on /proc (MS_NOSUID|MS_NODEV|MS_NOEXEC "")...
Bind-mounting /proc/sys on /proc/sys (MS_BIND "")...
(sd-namespace) succeeded.
Init process invoked as PID 5907
Bind-mounting /proc/sys/net on /proc/sys/net (MS_BIND "")...
Remounting /proc/sys (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /run/systemd/inaccessible/reg on /proc/kallsyms (MS_BIND "")...
Remounting /proc/kallsyms (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /run/systemd/inaccessible/reg on /proc/kcore (MS_BIND "")...
Remounting /proc/kcore (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /run/systemd/inaccessible/reg on /proc/keys (MS_BIND "")...
Remounting /proc/keys (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /run/systemd/inaccessible/reg on /proc/sysrq-trigger (MS_BIND "")...
Remounting /proc/sysrq-trigger (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /run/systemd/inaccessible/reg on /proc/timer_list (MS_BIND "")...
Remounting /proc/timer_list (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /proc/acpi on /proc/acpi (MS_BIND "")...
Remounting /proc/acpi (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Failed to create directory /proc/apm: No such file or directory
Failed to create directory /proc/apm: No such file or directory
Bind-mounting /proc/asound on /proc/asound (MS_BIND "")...
Remounting /proc/asound (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /proc/bus on /proc/bus (MS_BIND "")...
Remounting /proc/bus (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /proc/fs on /proc/fs (MS_BIND "")...
Remounting /proc/fs (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /proc/irq on /proc/irq (MS_BIND "")...
Remounting /proc/irq (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /proc/scsi on /proc/scsi (MS_BIND "")...
Remounting /proc/scsi (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Mounting mqueue on /dev/mqueue (MS_NOSUID|MS_NODEV|MS_NOEXEC "")...
Mounting sysfs on /sys/full (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC "")...
Bind-mounting /sys/full/block on /sys/block (MS_BIND "")...
Remounting /sys/block (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /sys/full/bus on /sys/bus (MS_BIND "")...
Remounting /sys/bus (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /sys/full/class on /sys/class (MS_BIND "")...
Remounting /sys/class (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /sys/full/dev on /sys/dev (MS_BIND "")...
Remounting /sys/dev (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /sys/full/devices on /sys/devices (MS_BIND "")...
Remounting /sys/devices (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Bind-mounting /sys/full/kernel on /sys/kernel (MS_BIND "")...
Remounting /sys/kernel (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND "")...
Umounting /sys/full...
Network interface dcn-ve2-f is not initialized yet.
```

**Comment by yuwata** [comment]
For some reason, `udevd` has not initialized the interface yet. Please call `udevadm control --log-priority=debug`, then re-create the veth interface. And please paste the logs obtained by `journalctl -u systemd-udevd.service`.

**Comment by DaanDeMeyer** [comment]
Turns out `udev` wasn't installed (and `systemd-udevd` wasn't running because of that). However, I'm running systemd-nspawn from within a podman container and starting `systemd-udevd` is failing because /sys/ isn't writable.  I'll try to fix that but I'm wondering if we could get this to work without udev? Is there a requirement for a network device to be registered with udev to move it to the container's network namespace?

`--network-veth-extra` is working without any issues which further makes me think this might be possible without udev.

**Comment by yuwata** [comment]
Fix is waiting in #14229. If possible, please test the PR. Thank you.
